lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1275,22 +1275,37 @@ def get_recommended_datastore(self, datastore_cluster_obj=None):
         Returns: Name of recommended datastore from the given datastore cluster
 
         """
+        datastore = None
         if datastore_cluster_obj is None:
-            return None
-        pod_sel_spec = vim.storageDrs.PodSelectionSpec()
-        pod_sel_spec.storagePod = datastore_cluster_obj
-        storage_spec = vim.storageDrs.StoragePlacementSpec()
-        storage_spec.podSelectionSpec = pod_sel_spec
-        storage_spec.type = 'create'
+            return datastore
+        # Check if Datastore Cluster provided by user is SDRS ready
+        sdrs_status = datastore_cluster_obj.podStorageDrsEntry.storageDrsConfig.podConfig.enabled
+        if sdrs_status:
+            # We can get storage recommendation only if SDRS is enabled on given datastorage cluster
+            pod_sel_spec = vim.storageDrs.PodSelectionSpec()
+            pod_sel_spec.storagePod = datastore_cluster_obj
+            storage_spec = vim.storageDrs.StoragePlacementSpec()
+            storage_spec.podSelectionSpec = pod_sel_spec
+            storage_spec.type = 'create'
 
-        try:
-            rec = self.content.storageResourceManager.RecommendDatastores(storageSpec=storage_spec)
-            rec_action = rec.recommendations[0].action[0]
-            real_datastore_name = rec_action.destination.name
-        except Exception as e:
-            # There is some error so we fall back to general workflow
-            return None
-        return real_datastore_name
+            try:
+                rec = self.content.storageResourceManager.RecommendDatastores(storageSpec=storage_spec)
+                rec_action = rec.recommendations[0].action[0]
+                return rec_action.destination.name
+            except Exception as e:
+                # There is some error so we fall back to general workflow
+                pass
+
+        datastore_freespace = 0
+        for ds in datastore_cluster_obj.childEntity:
+            if (isinstance(ds, vim.Datastore) and
+                    ((ds.summary.freeSpace > datastore_freespace) or (ds.summary.freeSpace == datastore_freespace and not datastore))):
```

Comment:
I don't think `or (ds.summary.freeSpace == datastore_freespace and not datastore)` is relevant.
The only case it handle is when all datastore in the datacenter have no free space left (freeSpace == 0) ⇒ in that case, I think it's better if the function return None as there is no "recommended" datastore.

Removing this check also make the code more understandable.

@Akasurde What do you think ?